File: framework/languages/node/samples/SupportChat.Ts/Server/Support/Infrastructure/ZLink/Spots/Handlers/close-conversation-handler.ts
import { zlinkSpotActorRequestHandler } from '@zlink-systems/nestjs';
import { ConversationSpot } from '../conversation-spot';
import { SupportUserActor } from '../../Actors/support-user-actor';
import { PacketNames } from '../../../../../../Shared/Contracts/messages';
import type {
  ZLinkSpotActorRequestContext,
  ZLinkSpotActorRequestHandler
} from '@zlink-systems/framework';
import type { ConversationSpot as ConversationSpotType } from '../conversation-spot';
import type { SupportUserActor as SupportUserActorType } from '../../Actors/support-user-actor';
import type {
  CloseConversationReq,
  CloseConversationRes
} from '../../../../../../Shared/Contracts/messages';

@zlinkSpotActorRequestHandler({
  actor: () => SupportUserActor,
  packetName: PacketNames.closeConversationReq,
  spot: () => ConversationSpot
})
class CloseConversationHandler
  implements ZLinkSpotActorRequestHandler<ConversationSpotType, SupportUserActorType, CloseConversationReq, CloseConversationRes> {
  async handle(
    spot: ConversationSpotType,
    actor: SupportUserActorType,
    context: ZLinkSpotActorRequestContext,
    request: CloseConversationReq
  ): Promise<CloseConversationRes> {
    void context;
    return await spot.close(actor, request);
  }
}

export { CloseConversationHandler };

File: framework/languages/java/samples/java/SupportChat/Server/Support/src/main/java/systems/zlink/samples/supportchat/server/support/handlers/CloseConversationHandler.java
package systems.zlink.samples.supportchat.server.support.handlers;

import systems.zlink.framework.channels.ZLinkRequestContext;
import systems.zlink.framework.channels.ZLinkRequestHandler;
import systems.zlink.framework.handlers.ZLinkHandlerGroup;
import systems.zlink.samples.supportchat.server.configuration.SampleNames;
import systems.zlink.samples.supportchat.server.support.domain.ConversationStore;
import systems.zlink.samples.supportchat.shared.contracts.Messages;

@ZLinkHandlerGroup(SampleNames.SupportChannel)
public final class CloseConversationHandler
    implements ZLinkRequestHandler<Messages.CloseConversationSupportReq, Messages.CloseConversationRes> {
    private final ConversationStore store;

    public CloseConversationHandler(ConversationStore store) {
        this.store = store;
    }

    @Override
    public Messages.CloseConversationRes handle(
        Messages.CloseConversationSupportReq request,
        ZLinkRequestContext context) {
        return store.close(request);
    }
}
